src/msg/xio/XioPool.h

```diff
@@ -194,16 +200,33 @@ extern XioPoolStats xp_stats;
 static inline int xpool_alloc(struct xio_mempool *pool, uint64_t size,
 			      struct xio_mempool_obj* mp)
 {
+  // try to allocate from the xio pool
+  int r = xio_mempool_alloc(pool, size, mp);
+  if (r == 0) {
```

Comment:
Maybe it's a naive question, why we will failed to alloc from xio mempool? AFAR, accelio could extend new memory block and return new alloced memory inline?